detector/data_reader.py

```diff
@@ -0,0 +1,52 @@
+import torchvision
+import os
+import torch
+import pandas as pd
+
+
+def create_label_list(data_path):
+    list_names = []
+    list_target = []
+    for f in os.listdir(data_path + "\\good"):
+        list_names.append("\\good\\" + f)
+        list_target.append(0)
+    for f in os.listdir(data_path + "\\defect"):
+        list_names.append("\\defect\\" + f)
+        list_target.append(1)
+
+    df1 = pd.DataFrame(list_names, columns=["file_name"])
+    df1["label"] = pd.DataFrame(list_target)
+    return df1
```

Comment:
Probably a matter of taste, but I think this is a little bit cleaner :)

```suggestion
    path_good = os.listdir(data_path + "\\good")
    df_good = pd.Dataframe({"file_name": [f"\\good\\{fn}" for fn in path_good]})
    df_good["label"] = 0
    path_bad = os.listdir(data_path + "\\badd")
    df_bad = pd.Dataframe({"file_name": [f"\\bad\\{fn}" for fn in path_bad]})
    df_bad["label"] = 1
    df_labels =  pd.concat([df_good, df_bad], axis=0)
    return df_labels
```

Reply:
Adapted to new code